pymbolic/mapper/unifier.py

```diff
@@ -247,28 +273,28 @@ def map_lookup(self,
 
     @override
     def map_sum(self,
-                expr: (p.Sum | p.Product
+                expr: (
+                    p.Sum | p.Product
                     | p.BitwiseAnd | p.BitwiseOr | p.BitwiseXor
                     | p.LogicalAnd | p.LogicalOr
                     | p.Min | p.Max),
                 other: Expression,
-                urecs: Sequence[UnificationRecord]):
+                urecs: Sequence[UnificationRecord]) -> Sequence[UnificationRecord]:
         if (not isinstance(other, type(expr))
                 or len(expr.children) != len(other.children)):
             return []
 
-        result = []
+        result: list[UnificationRecord] = []
 
         from pytools import generate_permutations
         had_structural_match = False
-        for perm in generate_permutations(range(len(expr.children))):
+        for perm in generate_permutations(list(range(len(expr.children)))):
```

Comment:
Is this used anywhere? `generate_permutations` doesn't work with iterables at all :grin:

Reply:
Hm, actually loopy uses that `UnidirectionalUnifier`, which overrides this `map_sum` in a different way, so I guess that's the one that actually runs.